infra/blueprint-test/modules/tft/terraform.go

```diff
@@ -0,0 +1,248 @@
+	return tft
+}
+
+// getTFModuleFixtureDir gets TF fixture config path from a fixture name.
+func getTFModuleFixtureDir(fixture string) string {
+	return fmt.Sprintf("../../test/fixtures/%s", fixture)
+}
+
+// getTFModuleSetupDir returns well known setup path.
+func getTFModuleSetupDir() string {
+	return "../../setup"
+}
+
+// getTFOptions generates terraform.Options used by Terratest.
+func (b *TFBlueprintTest) getTFOptions() *terraform.Options {
+	return terraform.WithDefaultRetryableErrors(b.t, &terraform.Options{
+		TerraformDir: b.tfDir,
+		EnvVars:      b.envVars,
+		Vars:         b.vars,
+		Logger:       b.logger,
+	})
+}
+
+// getTFSetupOPMap computes a map of TF outputs from setup.
+// Currently only returns string outputs.
+func (b *TFBlueprintTest) getTFSetupOPMap() map[string]string {
+	o := terraform.OutputAll(b.t, &terraform.Options{TerraformDir: b.setupPath, Logger: b.logger})
+	n := make(map[string]string)
+	for k, v := range o {
+		s, ok := v.(string)
+		if !ok {
+			b.logger.Logf(b.t, "Unable to convert output %s value in setup dir to string", k)
+			continue
+		}
+		n[k] = s
+	}
+	return n
+}
+
+// GetStringOutput returns TF output for a given key as string.
+// It fails test if given key does not output a primitive.
+func (b *TFBlueprintTest) GetStringOutput(name string) string {
+	return terraform.Output(b.t, b.getTFOptions(), name)
+}
+
+// GetTFSetupOPListVal returns TF output from setup for a given key as list.
+// It fails test if given key does not output a list type.
+func (b *TFBlueprintTest) GetTFSetupOPListVal(key string) []string {
+	if b.setupPath == "" {
+		b.t.Fatal("Setup path not set")
+	}
+	return terraform.OutputList(b.t, &terraform.Options{TerraformDir: b.setupPath, Logger: b.logger}, key)
+}
+
+// loadTFEnvVar adds new env variables prefixed with TF_VAR_ to an existing map of variables.
+func loadTFEnvVar(m map[string]string, new map[string]string) {
+	for k, v := range new {
+		m[fmt.Sprintf("TF_VAR_%s", k)] = v
+	}
+}
+
+// AutoDiscoverAndTest discovers TF config from examples/fixtures and runs tests.
+func AutoDiscoverAndTest(t *gotest.T) {
+	configs := discovery.FindTestConfigs(t, "./")
+	for _, dir := range configs {
+		// dir must be of the form ../fixture/name or ../examples/name
+		testName := fmt.Sprintf("test-%s-%s", path.Base(path.Dir(dir)), path.Base(dir))
+		t.Run(testName, func(t *gotest.T) {
+			nt := Init(t, WithTFDir(dir), WithSetupPath("../setup"))
+			bpt.TestBlueprint(t, nt, nil)
+		})
+	}
+}
+
+// Teardown runs TF destroy on a blueprint.
+func (b *TFBlueprintTest) Teardown() {
+	terraform.Destroy(b.t, b.getTFOptions())
+}
+
+// Verify asserts no resource changes exist after apply.
+func (b *TFBlueprintTest) Verify(assert *assert.Assertions) {
+	e := terraform.PlanExitCode(b.t, b.getTFOptions())
+	// exit code 0 is success with no diffs, 2 is success with non-empty diff
+	// https://www.terraform.io/docs/cli/commands/plan.html#detailed-exitcode
```

Comment:
We should test these separately: first assert it's not 1 (failure) then assert it's not 2 (non-empty diff).

Reply:
done